esphome/components/modbus/modbus.cpp

```diff
@@ -133,38 +168,78 @@ bool Modbus::parse_modbus_byte_(uint8_t byte) {
     if (device->address_ == address) {
       // Is it an error response?
       if ((function_code & 0x80) == 0x80) {
-        ESP_LOGD(TAG, "Modbus error function code: 0x%X exception: %d", function_code, raw[2]);
-        if (waiting_for_response != 0) {
-          device->on_modbus_error(function_code & 0x7F, raw[2]);
+        uint8_t exception = raw[2];
+        ESP_LOGD(TAG, "Modbus error function code: 0x%X exception: %d", function_code, exception);
+        if (waiting_for_response_ == address) {
+          set_timeout("on_modbus_error", 0, [device, function_code, exception]() {
```

Comment:
> All uses of class members and member functions should be prefixed with `this->` to distinguish them from global functions/variables.

https://esphome.io/guides/contributing#codebase-standards
```suggestion
          this->set_timeout("on_modbus_error", 0, [device, function_code, exception]() {
```

We actually have another function for simply performing in the next loop:
```suggestion
          this->defer("on_modbus_error", [device, function_code, exception]() {
```
Internally its just set_timeout with a 0, but better for component code to use it instead.